File: src/management/client/src/models/AzureSubscriptionResources.ts
// -----------------------------------------------------------------------
// <copyright file="AzureSubscriptionResources.ts" company="Microsoft Corporation">
//        Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
// -----------------------------------------------------------------------

import AzureSubscription from './AzureSubscription';
import AzureResource from './AzureResource';

export default interface AzureSubscriptionResources {
    /**
     * The Azure subscription.
     */
    subscription: AzureSubscription;

    /**
     * The Azure subscription resources.
     */
    resources: ReadonlyArray<AzureResource>;
}

File: src/management/server/SmartSignalsManagementApi/Models/AzureSubscriptionResources.cs
//-----------------------------------------------------------------------
// <copyright file="AzureSubscriptionResources.cs" company="Microsoft Corporation">
//        Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

namespace Microsoft.Azure.Monitoring.SmartAlerts.Appliance.ManagementApi.Models
{
    using System.Collections.Generic;
    using Microsoft.Azure.Monitoring.SmartAlerts.Clients;
    using Newtonsoft.Json;

    /// <summary>
    /// This class represents the model of the results for the GET /resource operation.
    /// </summary>
    public class AzureSubscriptionResources
    {
        /// <summary>
        /// Gets or sets the Azure subscription
        /// </summary>
        [JsonProperty("subscription")]
        public AzureSubscription Subscription { get; set; }

        /// <summary>
        /// Gets or sets the Azure subscription resources
        /// </summary>
        [JsonProperty("resources")]
        public List<ResourceIdentifier> Resources { get; set; }
    }
}
